src/components.jsx/Comment.jsx

```diff
@@ -33,9 +53,52 @@ const Comment = ( { comment } ) =>
               </Badge>
             </Box>
           </Flex>
+          {comment.author === user.username && (
+            <>
+              <DeleteIcon
+                key={commentId}
+                className="remove-comment"
+                ml="auto"
+                onClick={onOpen}
+              />
+              <AlertDialog
+                isOpen={isOpen}
+                leastDestructiveRef={cancelRef}
+                onClose={onClose}
+              >
+                <AlertDialogOverlay>
+                  <AlertDialogContent>
+                    <AlertDialogHeader fontSize="lg" fontWeight="bold">
+                      Delete Comment
+                    </AlertDialogHeader>
+
+                    <AlertDialogBody>
+                      Are you sure? You can't undo this action afterwards.
+                    </AlertDialogBody>
+
+                    <AlertDialogFooter>
+                      <Button ref={cancelRef} onClick={onClose}>
+                        Cancel
+                      </Button>
+                      <Button

```

Comment:
an Idea could be to abstract this logic and the DeleteButton just take a comment id